go/vt/vtgate/vindexes/geo_experimental.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vindexes
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/key"
+)
+
+var (
+	_ Vindex        = (*GeoExperimental)(nil)
+	_ Lookup        = (*GeoExperimental)(nil)
+	_ WantOwnerInfo = (*GeoExperimental)(nil)
+	_ MultiColumn   = (*GeoExperimental)(nil)
+)
+
+func init() {
+	Register("geo_experimental", NewGeoExperimental)
+}
+
+// GeoExperimental defines a vindex that uses a lookup table.
+// The table is expected to define the id column as unique. It's
+// Unique and a Lookup.
+type GeoExperimental struct {
+	regionBytes int
+	*ConsistentLookupUnique
+}
+
+// NewGeoExperimental creates a GeoExperimental vindex.
+// The supplied map requires all the fields of "consistent_lookup_unique".
+// Additionally, it requires a region_bytes argument whose value can be "1", or "2".
+func NewGeoExperimental(name string, m map[string]string) (Vindex, error) {
```

Comment:
I think we should rename it to `multi-region-vindex` or something that can convey what it will do clearly